src/templates/homepage/HeroSection.jsx

```diff
@@ -24,53 +24,59 @@ const HeroDropdownElem = ({ title }) => (
   </div>
 )
 
-const HeroDropdown = ({ title, options, isActive, toggleDropdown }) => (
-  <div
-    className={`bp-dropdown margin--top--sm ${isActive ? "is-active" : null}`}
-  >
-    <div className="bp-dropdown-trigger">
-      <a
-        className="bp-button bp-dropdown-button hero-dropdown is-centered"
-        role="button"
-        tabIndex="0"
-        aria-haspopup="true"
-        aria-controls="hero-dropdown-menu"
-        onClick={toggleDropdown}
-        onKeyDown={toggleDropdown}
-      >
-        <span>
-          <b>
-            <p>{title}</p>
-          </b>
-        </span>
-        <span className="icon is-small">
-          <i
-            className="sgds-icon sgds-icon-chevron-down is-size-4"
-            aria-hidden="true"
-          />
-        </span>
-      </a>
-    </div>
+const HeroDropdown = ({ title, options }) => {
+  const [dropdownIsActive, setDropdownIsActive] = useState(false)
+  const toggleDropdown = () => setDropdownIsActive((prevState) => !prevState)
+  return (
     <div
-      className="bp-dropdown-menu has-text-left"
-      id="hero-dropdown-menu"
-      role="menu"
+      className={`bp-dropdown margin--top--sm ${
+        dropdownIsActive ? "is-active" : null
+      }`}
     >
-      <div className="bp-dropdown-content is-centered">
-        {options
-          ? options.map((option, index) =>
-              option.title ? (
-                <HeroDropdownElem
-                  key={`dropdown-${index}`}
-                  title={option.title}
-                />
-              ) : null
-            )
-          : null}
+      <div className="bp-dropdown-trigger">
+        <a
+          className="bp-button bp-dropdown-button hero-dropdown is-centered"
+          role="button"
+          tabIndex="0"
+          aria-haspopup="true"
+          aria-controls="hero-dropdown-menu"
+          onClick={toggleDropdown}
+          onKeyDown={toggleDropdown}
+        >
+          <span>
+            <b>
+              <p>{title}</p>
+            </b>
+          </span>
+          <span className="icon is-small">
+            <i
+              className="sgds-icon sgds-icon-chevron-down is-size-4"
+              aria-hidden="true"
+            />
+          </span>
+        </a>
+      </div>
+      <div
+        className="bp-dropdown-menu has-text-left"
+        id="hero-dropdown-menu"
+        role="menu"
+      >
+        <div className="bp-dropdown-content is-centered">
+          {options
+            ? options.map((option, index) =>
+                option.title ? (
+                  <HeroDropdownElem
+                    key={`dropdown-${index}`}
+                    title={option.title}
+                  />
+                ) : null
+              )
+            : null}
```

Comment:
i think it's not ideal for us to have nested ternaries as it impacts readability. i'm not sure what's the best option here, i usually just write a small component at the top and use it here but maybe a better solution is to just ensure that the stuff passed in is not null (not really sure how to do this)

Reply:
[a887d80](https://github.com/isomerpages/isomercms-frontend/pull/804/commits/a887d802425f5c2210d70d6be78d60cdc8c169b4) Good idea